Khp3 6789

gabriel090 opened the pull request:
Create hearing screening report

[MOH 419B Hearing Screening Register Register_Apr-2025(1).xls](https://github.com/user-attachments/files/19627320/MOH.419B.Hearing.Screening.Register.Register_Apr-2025.1.xls)

patryllus commented:
@gabriel090 Kindly include videos or screenshots for these report changes with data to make reviewing easy.

njorocs reviewed (changes requested):
Check referred from, amount, receipt number, remarks, 1st and second screening dates.

gabriel090 commented:
> Check referred from, amount, receipt number, remarks, 1st and second screening dates.

I have resolved this issue. Renamed the amount and receipt columns correctly. I have removed the 2nd screening date as it is missing in the form, and also the remark column removed as it is missing in the form.
![Screenshot from 2025-04-07 20-23-41](https://github.com/user-attachments/assets/cc2f1bd0-ca09-41ed-8a7f-85b2549c17b2)

njorocs commented on api/src/main/java/org/openmrs/module/kenyaemr/metadata/CommonMetadata.java:
```diff
@@ -169,7 +169,7 @@ public static final class _Form {
 		public static final String DERMATOLOGY_CLINICAL_FORM = "efa2f992-44af-487e-aaa7-c92813a34612";
 		public static final String UROLOGY_CLINICAL_FORM = "57df8a60-7585-4fc0-b51b-e10e568cf53c";
-		public static final String HEARING_SCREENING_CLINICAL_FORM = "270f388f-439f-476a-8919-8211f850d366";
+		public static final String HEARING_SCREENING_CLINICAL_FORM = "4b5f79f5-f6bf-4dc2-b5c3-f5d77506775c";
```
What is the source of this UUID?

gabriel090 replied:
> What is the source of this UUID?

Fixed by updating the uuid from BA hearing form

njorocs commented on api/src/main/java/org/openmrs/module/kenyaemr/reporting/builder/common/MOH419BReportBuilder.java:
```diff
@@ -49,11 +41,11 @@
     public static final String ENC_DATE_FORMAT = "yyyy/MM/dd";
     public static final String DATE_FORMAT = "dd/MM/yyyy";
-	private static final String SPECIAL_CLINIC = CommonMetadata._Form.ORTHOPAEDIC_CLINICAL_FORM;
+	private static final String SPECIAL_CLINIC = CommonMetadata._Form.HEARING_SCREENING_CLINICAL_FORM;
```
Since you are dealing with the same special clinic in this class, it would be better to have a more intuitive name that communicates what special clinic we are handling.

gabriel090 replied:
Fixed

njorocs reviewed (changes requested):
Do not remove columns from the standard template. If missing in the form, just don't map them.

gabriel090 commented:
> Do not remove columns from the standard template. If missing in the form, just don't map them.

Worked on this

njorocs reviewed (approved):
Looks okay
